pkgs/development/compilers/jwasm/default.nix

```diff
@@ -3,15 +3,15 @@
 , fetchFromGitHub
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "jwasm";
-  version = "2.15";
+  version = "2.16";
 
   src = fetchFromGitHub {
     owner = "Baron-von-Riedesel";
     repo  = "JWasm";
-    rev = "v${version}";
-    hash = "sha256-ef4uEtEpnqYGhFmxuefJ40zyOuHsiPOLpH/52i7a7KI=";
+    rev = "v${finalAttrs.version}";
```

Comment:
May I ask, what is the point of allowing to override the rev here? I understand that `finalAttrs` allows for finer overriding of the attributes, but if I specify another rev, then it would fetch a different git commit, which would result in a different sha256 sum anyway.

Reply:
This is not about overriding per se, but about getting rid of `rec` keyword, as recommended by [Nix Dev Guide](https://nix.dev/) on its [Anti-Patterns](https://nix.dev/anti-patterns/language#rec-expression).

Reply:
I still fail to understand. It lists two pitfalls. Potential infinite recursion and surprising `overrideAttrs` behavior. Neither of which affects us in a case like this. 

I am asking because I would also do this kind of changes during my contributions, but I fail to see the advantage of it.